monai/networks/nets/torchvision_fc.py

```diff
@@ -9,18 +9,68 @@
 # limitations under the License.
 
-from typing import Tuple, Union
+from typing import Any, Dict, Optional, Tuple, Union
 
 import torch
 
-from monai.utils import optional_import
+from monai.networks.nets import NetAdapter
+from monai.utils import deprecated, optional_import
 
 models, _ = optional_import("torchvision.models")
 
 
-__all__ = ["TorchVisionFullyConvModel"]
+__all__ = ["TorchVisionFCModel", "TorchVisionFullyConvModel"]
 
 
+class TorchVisionFCModel(NetAdapter):
+    """
+    Customize the fully connected layer of TorchVision model or replace it by convolutional layer.
+
+    Args:
+        model_name: name of any torchvision model with fully connected layer at the end.
+            ``resnet18`` (default), ``resnet34m``, ``resnet50``, ``resnet101``, ``resnet152``,
+            ``resnext50_32x4d``, ``resnext101_32x8d``, ``wide_resnet50_2``, ``wide_resnet101_2``.
+            model details: https://pytorch.org/vision/stable/models.html.
+        n_classes: number of classes for the last classification layer. Default to 1.
+        dim: number of spatial dimensions, default to 2.
+        in_channels: number of the input channels of last layer. if None, get it from `in_features` of last layer.
+        use_conv: whether use convolutional layer to replace the last layer, default to False.
+        pool: parameters for the pooling layer, it should be a tuple, the first item is name of the pooling layer,
+            the second item is dictionary of the initialization args. if None, will not replace the `layers[-2]`.
+            default to `("avg", {"kernel_size": 7, "stride": 1})`.
+        bias: the bias value when replacing the last layer. if False, the layer will not learn an additive bias,
+            default to True.
+        pretrained: whether to use the imagenet pretrained weights. Default to False.
+    """
+
+    def __init__(
+        self,
+        model_name: str = "resnet18",
+        n_classes: int = 1,
+        dim: int = 2,
+        in_channels: Optional[int] = None,
+        use_conv: bool = False,
+        pool: Optional[Tuple[str, Dict[str, Any]]] = ("avg", {"kernel_size": 7, "stride": 1}),
+        bias: bool = True,
+        pretrained: bool = False,
+    ):
+        model = getattr(models, model_name)(pretrained=pretrained)
+        # check if the model is compatible, should have a FC layer at the end
+        if not str(list(model.children())[-1]).startswith("Linear"):
+            raise ValueError(f"Model ['{model_name}'] does not have a Linear layer at the end.")
+
+        super().__init__(
+            model=model,
+            n_classes=n_classes,
+            dim=dim,
+            in_channels=in_channels,
+            use_conv=use_conv,
+            pool=pool,
+            bias=bias,
+        )
+
+
+@deprecated(since="0.6.0", version_val="0.7.0", msg_suffix="please consider to use `TorchVisionFCModel` instead.")
 class TorchVisionFullyConvModel(torch.nn.Module):
```

Comment:
could implement the current `TorchVisionFullyConvModel ` with the new `TorchVisionFCModel` class? it's also to test that the new API is more flexible

Reply:
Hi @wyli ,

Sure, updated to use the new `TorchVisionFCModel` class in the latest commit.
Could you please help review it again?

Thanks in advance.